src/main/java/org/apache/mesos/dcos/DcosCluster.java

```diff
@@ -1,19 +1,28 @@
 package org.apache.mesos.dcos;
 
-import org.apache.http.client.fluent.Content;
-import org.apache.http.client.fluent.Request;
+import org.apache.commons.io.Charsets;
 import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Optional;
 
 /**
  * Instances of this class represent DC/OS clusters.
  */
 public class DcosCluster {
-    private static final String DCOS_VERSION_PATH = "/dcos-metadata/dcos-version.json";
+    private static final Logger LOGGER = LoggerFactory.getLogger(DcosCluster.class);
+
+    /**
+     * Location of file on DC/OS cluster that can be used to extract version information.
+     */
+    public static final String DCOS_VERSION_PATH = "/opt/mesosphere/active/dcos-metadata/etc/dcos-version.json";
```

Comment:
@mohitsoni I think what @triclambert is saying is that in general we're not supposed to access anything `/opt/mesosphere` according to the Foundations team.  Of course we violate this for at least `/opt/mesosphere/libmesos.so`.  This change was originally intended to avoid problems reaching the endpoint in strict mode clusters right?

Maybe we should still attempt to hit the HTTP endpoint and degrade gracefully, instead of looking for a local file...